### PR TITLE
Adding unit checks to the public API

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Currently, we are not enforcing units at several public interfaces. This PR aims to fix this and enforces the internal use of the openmm unit system. This PR will close issues #54 and #32.
The `forward` call of the models will check for units attached to `positions` and convert them to nanometers if possible. If no units are attached, it will assume that the units are in nanometers and print a warning to stderr.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Refactor all public APIs to ensure that we check and validate units
  - [x] Conversion to openmm unit system

## Status
- [x] Ready to go

### CODE REVIEW

File: modelforge/tests/test_utils.py [hunk 1]
```diff
@@ -89,13 +97,15 @@ def test_gaussian_rbf(RBF):
     assert expected_output.shape == (3, n_rbf)
 
 
-@pytest.mark.parametrize("RBF", [GaussianRBF])
+@pytest.mark.parametrize("RBF", [_GaussianRBF])
 def test_rbf_invariance(RBF):
     # Define a set of coordinates
-    coordinates = torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0]])
+    from openmm import unit
```
Comment (chrisiacovella) [inline modelforge/tests/test_utils.py:103]: Why are you importing openmm units here? 

We had been using openff units in the rest of the code.  Openff-units are directly compatible with openmm units.  `openff.units.openmm` has  `to_openmm` and  `from_openmm` functions if we do need to convert.

File: modelforge/tests/test_utils.py [hunk 2]
```diff
@@ -12,36 +12,39 @@ def test_cosine_cutoff():
...
-    expected_output = torch.tensor([0.5, 0.0, 0.0])
-    output = cosine_cutoff(d_ij, cutoff)
-    assert torch.allclose(output, expected_output, rtol=1e-3)
 
 
 def test_cosine_cutoff_module():
     # Test CosineCutoff module
-    d_ij = torch.tensor([1.0, 2.0, 3.0])
-    cutoff = 2.0
+    from openmm import unit
```
Comment (chrisiacovella) [inline modelforge/tests/test_utils.py:31]: I think we should be using openff-units.

File: modelforge/tests/test_utils.py [hunk 3]
```diff
@@ -73,10 +79,12 @@ def test_gaussian_rbf(RBF):
     assert gaussian_rbf.n_rbf == n_rbf
 
     # Test that the cutoff distance is correct
-    assert gaussian_rbf.cutoff == cutoff
+    assert gaussian_rbf.cutoff == cutoff.value_in_unit_system(unit.md_unit_system)
```
Comment (chrisiacovella) [inline modelforge/tests/test_utils.py:82]: I think we should use openff units, as we've bee using those throughout the code already. 

we would use the syntax:

`assert gaussian_rbf.cutoff == cutoff.is_compatible_with(unit.nanometer)`

**Comment by chrisiacovella** [comment]
We had been using openff-units already in the package.  I don't think we should switch to openmm-units for the public facing API. Openff-units should be more flexible  (as we can easily convert to/from openmm units if we need) and has a convenient function unit.parse_expression, to make it easily to reassign units from the HDF5 files (which are encoded as strings).  

I think we should actually probably switch to openff-units in Chiron, just doing conversions from the openmm test systems (but that is a conversation for a different day in a different repo).

**Review by chrisiacovella — CHANGES_REQUESTED** [review]
as I just commented, I think we should be using openff-units. Maybe we can discuss this quickly offline.  I'm happy to switch these top openff.

**Comment by chrisiacovella** [comment]
After quick discussion, we are going to use openff-units.  I'll be grabbing a copy of the branch and making these changes.

**Comment by wiederm** [comment]
Thanks @chrisiacovella for these changes!

Sorry for introducing yet another unit package! Thanks for catching that in time and making the necessary modifications!

File: modelforge/potential/utils.py [hunk 4]
```diff
@@ -172,28 +172,32 @@ def gaussian_rbf(
...
-        r"""
-        Behler-style cosine cutoff module.
+from openff.units import unit
 
-        Args:
-            cutoff (float): The cutoff distance.
 
-        Attributes:
-            cutoff (torch.Tensor): The cutoff distance as a tensor.
+class _CosineCutoff(nn.Module):
```
[the remark this answers is not in the document]
Comment (chrisiacovella) [reply modelforge/potential/utils.py:178]: Should this be a public class

**Review by chrisiacovella — APPROVED** [review]

**Pull request merged**